### DISCUSSION TITLE
Challenges and Help - Running Spezi on MacOS Ventura

### DISCUSSION THREAD

**Discussion by pstell2002** [StanfordSpezi/.github#68] [Challenges and Help] [answered]
### What Stanford Spezi module is your challenge related to?

SpeziTemplateApplication

### Description

I am trying to build the Spezi Template Application using my MacBook Pro 2017 and XCode version 15.2. I understand that this does not meet the software prerequisites and am facing the following error:

<img width="935" alt="Screenshot 2024-12-01 at 8 08 24 PM" src="https://github.com/user-attachments/assets/7aa9f118-08e5-44fe-a30d-21b4b7567b8c">

I have also tried downgrading the deployment target to IOS 17.0 but that resulted in failed builds with "Package.resolved file is corrupted or malformed" errors. 

I am wondering if there is a way to work on the Spezi Template Application without getting a new device. 

### Reproduction

Cloned template application repo into XCode and built project: 
https://github.com/StanfordBDHG/TemplateApplication/tree/main 

### Expected behavior

Template Application to build on a simulated or physical device (iPhone with IOS 18 installed) 

### Code of Conduct

- [X] I agree to follow this projects's Code of Conduct and Contributing Guidelines

**Comment by lukaskollmer** [comment]
You need to select a different (i.e., non-build-only) run destination in Xcode. You can do this via the "SpeziTemplateApplication > Any iOS Device" menu in the top center of the Xcode window. Select either a Simulator (you might need to install one if none are present) or a physical iOS device if you have access to one running a supported iOS version.

Regarding the package.resolved issues: try File → Packages → Reset Package Caches, and if that does not work, try deleting the SpeziTemplateApplication folder in`$USER/Library/Developer/Xcode/DerivedData`.

**Comment by pstell2002** [comment]
I have tried selecting a different run destination, but with the default settings this is what shows up, even when I try to run the project on my physical iPhone with IOS 18 installed, which should be a supported version. 
<img width="564" alt="Screenshot 2024-12-05 at 8 37 02 PM" src="https://github.com/user-attachments/assets/55408d4d-9ce0-4856-ad93-6cfe8b889b1e">
 
The device targets only show up when I downgrade the minimum deployment to IOS 17.2, but the build failes due to the package issues:

<img width="629" alt="Screenshot 2024-12-05 at 8 44 39 PM" src="https://github.com/user-attachments/assets/f64aaa28-88f3-4f0d-b44a-6e3a1b40dd15">

For the package.resolved issue, I have tried File → Packages → Reset Package Caches, then deleting the SpeziTemplateApplication folder in$USER/Library/Developer/Xcode/DerivedData. I have also tried this solution: https://stackoverflow.com/questions/67185817/package-resolved-file-is-corrupted-or-malformed

However, I still face the same issues:

<img width="299" alt="Screenshot 2024-12-05 at 8 53 54 PM" src="https://github.com/user-attachments/assets/38824b4f-04b8-4213-be31-eeae43799826">

**Reply by PSchmiedmayer** [reply, marked as answer]
@pstell2002 It looks like you don't have an iOS 18-related simulator installed that is causing the issues with your run destinations. Can you try to ensure that
1. The latest iOS 18.2 simulators are installed (https://developer.apple.com/documentation/xcode/installing-additional-simulator-runtimes).
2. You have installed Xcode 16.2 
3. If there is no Simulator available click on "Manage Run Destinations ..." when selecting the simulators & adding iOS 18.2 ones? 

Last, but not least, can you ensure that you have cloned the latest version of the template application including all PRs that have been merged in the recent months?

**Reply by PSchmiedmayer** [reply]
@pstell2002 Have you eventually resolved the issue?
